### PR TITLE
fix include sequence after pinocchio commit c1ff8821c437e2401b80b7aa7f3f118464e2c051

### PR DESCRIPTION
**Pull request by proyan**
Pinocchio commit c1ff8821c437e2401b80b7aa7f3f118464e2c051 forces pinocchio inclusion before boost.
So crocoddyl now fails to compile with pinocchio devel.
This PR is meant to change the header include order based on the following logic:

1) Non-boost system headers
2) pinocchio headers
3) crocoddyl headers
4) pinocchio binding headers
5) crocoddyl binding headers

### CODE REVIEW

File: bindings/python/crocoddyl/multibody/frames.cpp [hunk 1]
```diff
@@ -6,15 +6,17 @@
 // All rights reserved.
 ///////////////////////////////////////////////////////////////////////////////
 
+#include "crocoddyl/multibody/frames.hpp"
+
 #include <eigenpy/memory.hpp>
 #include <eigenpy/eigen-to-python.hpp>
 
+
+#include "pinocchio/bindings/python/utils/std-aligned-vector.hpp"
```
Comment (cmastalli) [inline bindings/python/crocoddyl/multibody/frames.cpp:15]: Could you use `<>` simbol (as it is an external header)?

File: include/crocoddyl/core/activation-base.hpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 // BSD 3-Clause License
 //
-// Copyright (C) 2019-2021, LAAS-CNRS, University of Edinburgh
+// Copyright (C) 2019-2020, LAAS-CNRS, University of Edinburgh
```
Comment (cmastalli) [inline include/crocoddyl/core/activation-base.hpp:4]: Please revert the change on the copyright year.

File: include/crocoddyl/core/activation-base.hpp [hunk 3]
```diff
@@ -12,12 +12,13 @@
 #include <stdexcept>
 #include <boost/shared_ptr.hpp>
 #include <boost/make_shared.hpp>
-#include <boost/core/demangle.hpp>
 
 #include "crocoddyl/core/fwd.hpp"
 #include "crocoddyl/core/mathbase.hpp"
 #include "crocoddyl/core/utils/to-string.hpp"
 
+#include <boost/core/demangle.hpp>
```
Comment (cmastalli) [inline include/crocoddyl/core/activation-base.hpp:20]: Why do we need to include this header after Crocoddyl stuff?

The common rule is to include external headers first.

File: include/crocoddyl/core/integrator/euler.hxx [hunk 4]
```diff
@@ -6,12 +6,12 @@
 // All rights reserved.
 ///////////////////////////////////////////////////////////////////////////////
 
+#include "crocoddyl/core/utils/exception.hpp"
```
Comment (cmastalli) [inline include/crocoddyl/core/integrator/euler.hxx:9]: Same here.

**Review by cmastalli — CHANGES_REQUESTED** [review]
Thanks Rohan for taking care of this tiny, yet important, details.

I have left a few comments which I believe they need to be addressed.

File: bindings/python/crocoddyl/multibody/frames.cpp (hunk 1 above)
Comment (proyan) [reply bindings/python/crocoddyl/multibody/frames.cpp]: sure

File: bindings/python/crocoddyl/multibody/frames.cpp [hunk 5]
```diff
@@ -6,15 +6,17 @@
 // All rights reserved.
 ///////////////////////////////////////////////////////////////////////////////
 
+#include "crocoddyl/multibody/frames.hpp"
```
[the remark this answers is not in the document]
Comment (proyan) [reply bindings/python/crocoddyl/multibody/frames.cpp:9]: eigenpy uses boost, so no. Similarly, pinocchio bindings use boost

File: include/crocoddyl/core/activation-base.hpp (hunk 3 above)
Comment (proyan) [reply include/crocoddyl/core/activation-base.hpp]: Sure

File: include/crocoddyl/core/integrator/euler.hxx (hunk 4 above)
Comment (proyan) [reply include/crocoddyl/core/integrator/euler.hxx]: Sure

**Comment by proyan** [comment]
Thanks for the review Carlos. This isn't a tiny detail, unfortunately. Crocoddyl devel doesn't compile with pinocchio devel because of the aforementioned commit. This is meant to fix that.

**Comment by cmastalli** [comment]
> Thanks for the review Carlos. This isn't a tiny detail, unfortunately. Crocoddyl devel doesn't compile with pinocchio devel because of the aforementioned commit. This is meant to fix that.

Thanks for the explanation, I wasn't aware.

**Review by cmastalli — APPROVED** [review]
All alright from my side. However, the code might need to be formatted.

**Comment by jcarpent** [comment]
> Thanks for the review Carlos. This isn't a tiny detail, unfortunately. Crocoddyl devel doesn't compile with pinocchio devel because of the aforementioned commit. This is meant to fix that.

I've removed some macro def which were not aimed to stay within Pinocchio. To avoid client breacks, I may restore one commit.

**Comment by proyan** [comment]
> I've removed some macro def which were not aimed to stay within Pinocchio. To avoid client breacks, I may restore one commit.

Ok, I think that would be better for other downstream packages. For now Ill merge this
